### PR TITLE
Items on display

### PR DESCRIPTION
**Pull request by diversifiedmechanic**
This is mostly setting up the logic for the images with some tinkering with getting ready for once my db is seeded correctly.

### CODE REVIEW

File: src/components/ItemContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+  const ArrowBuffer = styled.div`
+    height: 30%
+  `;
+
+  const ArrowContainer = styled.div`
+    display: flex;
+    flex-direction: column;
+  `;
+
```
Comment (naszzz) [inline src/components/ItemContainer.jsx]: isn't this so weird. it's been happening to me too where some of my letters are different colors than the rest (i.e. ArrowContainer).

File: src/components/ItemContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    box-shadow:
+      0 0 5px 5px rgb(0, 0, 0, 0.07),
+      0 0 0px 0px rgb(0, 0, 0, 0.07);
+    :hover {
+      box-shadow:
+        0 0 5px 5px rgb(0, 0, 0, 0.15),
+        0 0 0px 0px rgb(0, 0, 0, 0.15);
+      cursor: pointer;
+      background-color: #f9f9f9;
+    }
```
Comment (naszzz) [inline src/components/ItemContainer.jsx]: this is cool. i wanna try to use box-shadow. don't think i have used it yet...

File: src/components/ItemContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  `;
+
+  // eslint-disable-next-line consistent-return
```
Comment (naszzz) [inline src/components/ItemContainer.jsx]: // eslint-disable-next-line consistent-return ..... hahahaaa... nice!!! this LGTM

File: src/components/ItemEntry.jsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+const itemEntry = ({ item }) => {
+  const {
+    itemName = null,
+    itemPrice = null,
+    itemPicture = null,
+    freeShipping = 'yes',
+  } = item;
```
Comment (naszzz) [inline src/components/ItemEntry.jsx]: this is awesome.

File: src/components/ItemContainer.jsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+    for (let i = 0; i < classNameArray.length; i++) {
+      if (classNameArray[i] === 'left-arrow') {
+        return onArrowClickCallback(-1);
+      }
+      if (classNameArray[i] === 'right-arrow') {
+        return onArrowClickCallback(1);
+      }
+    }
+  };
```
Comment (naszzz) [inline src/components/ItemContainer.jsx]: never seen this before.. className is built in prop of e.target ? pretty cool.

File: src/components/App.jsx [hunk 6]
```diff
@@ -1,56 +1,110 @@
...
+      // reset the array after going right then going left before hitting the normal reset
+      const newAllItems = [...allItems].splice(0, allItems.length - 4);
+      this.setState({
+        allItems: newAllItems,
+        itemsToDisplayIndex: newAllItems.length - 5,
+        scrollDirection: '',
+      });
+    } else if (scrollDirection === 'left' && itemsToDisplayIndex === allItems.length - 8) {
+      // reset the array after 'going around' to the left
+      const newAllItems = [...allItems].splice(0, allItems.length - 4);
```
Comment (naszzz) [inline src/components/App.jsx]: it's a lot of code, but i like it. think the logic is pretty nifty.

File: src/components/ItemContainer.jsx (hunk 1 above)
Comment (diversifiedmechanic) [reply src/components/ItemContainer.jsx]: That is really strange.  It's only happening in GitHub though, which is good.

File: src/components/ItemContainer.jsx (hunk 2 above)
Comment (diversifiedmechanic) [reply src/components/ItemContainer.jsx]: It gives that 'halo' effect.  You can do some really sick stuff with it besides just making the background slightly grey lol

File: src/components/ItemContainer.jsx (hunk 3 above)
Comment (diversifiedmechanic) [reply src/components/ItemContainer.jsx]: .... hey.  I couldn't figure out how to get it to see my return statements inside my conditionals.  I didn't have time to figure it out the right way.  If you have suggestions, let me know!

File: src/components/App.jsx (hunk 6 above)
Comment (diversifiedmechanic) [reply src/components/App.jsx]: There has to be a way to bring the code down some.  It's kind of dumb but it works for now so I'm moving on!

**Pull request merged**